presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduSessionProperties.java

```diff
@@ -14,31 +14,53 @@
 package io.prestosql.plugin.kudu;
 
 import com.google.common.collect.ImmutableList;
+import io.airlift.units.Duration;
+import io.prestosql.spi.PrestoException;
 import io.prestosql.spi.connector.ConnectorSession;
 import io.prestosql.spi.session.PropertyMetadata;
 
 import javax.inject.Inject;
 
 import java.util.List;
 
+import static io.prestosql.spi.StandardErrorCode.INVALID_SESSION_PROPERTY;
 import static io.prestosql.spi.session.PropertyMetadata.booleanProperty;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static java.lang.String.format;
 
 public final class KuduSessionProperties
 {
     private static final String KUDU_GROUPED_EXECUTION_ENABLED = "grouped_execution";
+    private static final String DYNAMIC_FILTERING_WAIT_TIMEOUT = "dynamic_filtering_wait_timeout";
 
     private final List<PropertyMetadata<?>> sessionProperties;
 
     @Inject
     public KuduSessionProperties(KuduClientConfig kuduConfig)
     {
-        sessionProperties = ImmutableList.<PropertyMetadata<?>>builder()
-            .add(booleanProperty(
-                KUDU_GROUPED_EXECUTION_ENABLED,
-                "Enable grouped execution.",
-                kuduConfig.isGroupedExecutionEnabled(),
-                false))
-            .build();
+        sessionProperties = ImmutableList.of(
+            booleanProperty(
+                    KUDU_GROUPED_EXECUTION_ENABLED,
+                    "Enable grouped execution.",
+                    kuduConfig.isGroupedExecutionEnabled(),
+                    false),
+            new PropertyMetadata<>(
+                    DYNAMIC_FILTERING_WAIT_TIMEOUT,
+                    "Duration to wait for completion of dynamic filters",
+                    VARCHAR,
+                    Duration.class,
+                    kuduConfig.getDynamicFilteringWaitTimeout(),
+                    false,
+                    value -> {
+                        Duration duration = Duration.valueOf((String) value);
+                        if (duration.getValue() < 0) {
```

Comment:
I think in negative value case `Duration.valueOf` will throw exception anyway, we won't reach here

Reply:
Updated